### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by vsicurella**
Algorithm improvements, style fixes, other fixes.

Still need to do some reimplementation, which will be done through the model.

### CODE REVIEW

File: src/js/generators.js [hunk 1]
```diff
@@ -291,7 +291,7 @@ function generate_enumerate_chord() {
 
   if (isInversion) {
     debug("This is an inversion. Chord is " + chord);
-    chord = invert_chord(chord);
+    chord = invertChord(chord);
```
Comment (meszaros-lajos-gyorgy) [inline src/js/generators.js]: nice one!

File: src/js/modifiers.js [hunk 2]
```diff
@@ -367,34 +368,34 @@ function modify_replace_with_approximation () {
   const tuning_table = model.get('tuning table')
   const newline = model.get('newline') === 'windows' ? WINDOWS_NEWLINE : UNIX_NEWLINE
 
-  var degree_selected = parseInt(jQuery( "#input_scale_degree" ).val());
+  var degreeSelected = parseInt(jQuery( "#input_scale_degree" ).val());
```
Comment (meszaros-lajos-gyorgy) [inline src/js/modifiers.js]: This should definitely be loaded from the model

File: src/js/modifiers.js [hunk 3]
```diff
@@ -433,17 +434,19 @@ function modify_update_approximations() {
     if (maxcentsd < 0)
       maxcentsd = 0;
 
-    var menulength = (semiconvergents) ? current_approximations.ratios.length : current_approximations.convergent_indicies.length;
+    var menulength = (semiconvergents) ? currentRatioStructure.length : currentRatioStructure.cf.length;
```
Comment (meszaros-lajos-gyorgy) [inline src/js/modifiers.js]: why do we need the parenthesis around semiconvergents?

File: src/js/scaleworkshop.js [hunk 4]
```diff
@@ -18,7 +18,6 @@ import {
   line_to_decimal,
   sanitize_filename,
 } from './helpers/converters.js'
-import { synth } from './synth.js'
```
Comment (meszaros-lajos-gyorgy) [inline src/js/scaleworkshop.js]: These were already removed in the latest `import-export` commit, please check, if you have the latest version merged on your branch.

File: src/js/modifiers.js (hunk 3 above)
Comment (vsicurella) [reply src/js/modifiers.js]: Will clean this whole function up when I update it to use the model

File: src/js/scaleworkshop.js (hunk 4 above)
Comment (vsicurella) [reply src/js/scaleworkshop.js]: I believe this was caused by a bad merge on my part, sorry about that!

**Pull request merged**